codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -193,6 +193,7 @@ public ChannelFuture writePriority(ChannelHandlerContext ctx, int streamId, int
             // The set priority operation must be done before sending the frame. The parent may not yet exist
             // and the priority tree may also be modified before sending.
             stream.setPriority(streamDependency, weight, exclusive);
+            stream.frameSent();
```

Comment:
It seems like we may be allowing a RST_STREAM frame to be sent on a stream in the IDLE state.  What if this stream was created in the IDLE state, we mark `frameSent()`, then the user tries to send a RST_STREAM frame we will send it, and the peer should close the connection with a PROTOCOL_ERROR via the spec:

https://tools.ietf.org/html/rfc7540#section-6.4:

> RST_STREAM frames MUST NOT be sent for a stream in the "idle" state.
   If a RST_STREAM frame identifying an idle stream is received, the
   recipient MUST treat this as a connection error (Section 5.4.1) of
   type PROTOCOL_ERROR.